corehq/apps/ivr/migrations/0002_call_app_id.py

```diff
@@ -0,0 +1,45 @@
+from django.db import migrations, models
+
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+def _update_model(model, instance, domain_forms):
+    domain = instance.domain
+    if domain not in domain_forms:
+        apps = get_apps_in_domain(domain)
+        domain_forms[domain] = {}
+        for app in apps:
+            for module in app.modules:
+                for form in module.get_forms():
+                    domain_forms[domain][form.unique_id] = app.get_id
+
+    app_id = domain_forms[domain].get(instance.form_unique_id, None)
+    if app_id:
+        model.objects.filter(form_unique_id=instance.form_unique_id).update(app_id=app_id)
```

Comment:
is it possible that some of these models will be left null. Will you be checking these records once the data has been migrated? If that is possible, what's the failure mode? Are the records essentially lost forever?

Reply:
It's possible in cases like the form_unique_id was set on the model and that app was later deleted, but those situations would already be failing on the `get_form` call, since `xforms_index` is populated by looping over `Application` docs: https://github.com/dimagi/commcare-hq/blob/96b79d77f60d9569a9a7bd2d183177ce267e5800/corehq/apps/app_manager/_design/views/xforms_index/map.js#L12-L21